src/ring_theory/roots_of_unity.lean

```diff
@@ -581,6 +618,88 @@ begin
     exact ⟨i, ⟨hin, hi.symm⟩, H⟩ }
 end
 
+/-The sets `primitive_roots k R` are pairwise disjoint. -/
+lemma prim_roots_disjoint {k l : ℕ} (hk : 0 < k) (hl : 0 < l) (h : k ≠ l) :
+  disjoint (primitive_roots k R) (primitive_roots l R) :=
+begin
+  intros z hz,
+  simp only [finset.inf_eq_inter, finset.mem_inter] at hz,
+  have hkl : k = l,
+  { apply nat.dvd_antisymm,
+    { have hpowk := ((is_primitive_root.iff_def z l).1 ((mem_primitive_roots hl).1 hz.2)).1,
+      exact ((is_primitive_root.iff_def z k).1 ((mem_primitive_roots hk).1 hz.1)).2 l hpowk },
+    { have hpowl := ((is_primitive_root.iff_def z k).1 ((mem_primitive_roots hk).1 hz.1)).1,
+      exact ((is_primitive_root.iff_def z l).1 ((mem_primitive_roots hl).1 hz.2)).2 k hpowl } },
+  exfalso,
+  exact h hkl
+end
+
+/-If in `R` there is a `n`-th primitive root of unity and `b` divides `n`,
+then there is in `R` a `b`-th primitive root of unity. -/
+lemma prim_of_pow_prim {ζ : R} {n : ℕ+} {a b : ℕ} (h : is_primitive_root ζ n) (hprod : ↑n = a * b)
```

Comment:
This should go in the `is_primitive_root` namespace: `is_primitive_root.pow` to enable dot notation.
Also, I think it might be better to have `n : nat` instead of `n : pnat`, and then add the condition `0 < a`.

Reply:
It now has `n : nat` instead of `n : pnat`.

It already is in the `is_primitive_root`, isn't it? Or you mean that it's better to change is name?

Reply:
Right, I meant the name change

Reply:
So you suggest to call it simply `pow` (plus the namespace of course)?

Reply:
Yup.

Reply:
Done.